src/main/java/com/artipie/asto/fs/RxFile.java

```diff
@@ -108,24 +108,26 @@ public RxFile(final Path file, final FileSystem fls) {
      * @return Completion or error signal
      */
     public Completable save(final Flowable<ByteBuffer> flow) {
-        return Single.fromCallable(
-            () -> FileChannel.open(
-                this.file,
-                StandardOpenOption.CREATE,
-                StandardOpenOption.WRITE,
-                StandardOpenOption.TRUNCATE_EXISTING
-            )
-        ).flatMapCompletable(
-            chan -> flow.reduce(
-                chan,
-                (out, buf) -> {
-                    while (buf.hasRemaining()) {
-                        out.write(buf);
+        return this.fls.rxOpen(
+            this.file.toString(),
+            new OpenOptions()
+                .setRead(false)
+                .setCreate(true)
+                .setWrite(true)
+                .setTruncateExisting(true)
+        )
+            .flatMapCompletable(
+                asyncFile -> Completable.create(
+                    emitter -> {
+                        flow.map(buf -> Buffer.buffer(new Remaining(buf).bytes()))
+                            .subscribe(asyncFile.toSubscriber()
+                                .onWriteStreamEnd(emitter::onComplete)
+                                .onWriteStreamError(emitter::onError)
```

Comment:
@Sammers21 what's the difference between `onWriteStreamError` and `onError`? Don't we need to wrap stream errors somehow to distinguish from just errors? E.g.:
```java
onWriteStreamError(err -> emitter.onError(new IOException("AsyncFile stream error", err)))
```